### PR TITLE
fix: JOIN-37218 update pubsub lib with processing of nullable arrays

### PR DESCRIPTION
**Pull request by eugeneswave**

### CODE REVIEW

File: packages/pubsub/src/__tests__/Subscriber.test.ts [hunk 1]
```diff
@@ -415,6 +423,50 @@ describe('Subscriber', () => {
...
+
+    it('receives avro parsed data and replaces 2 empty array with undefined using path from metadata', async () => {
+      topicMock.exists.mockResolvedValue([true])
+      subscriptionMock.exists.mockResolvedValue([true])
+      topicMock.getMetadata.mockResolvedValue([{'schemaSettings': {'schema': 'mock-schema'}}])
+      schemaClientMock.getSchema.mockResolvedValue([{definition: JSON.stringify(SCHEMA_DEFINITION_READER_OPTIONAL_ARRAY_EXAMPLE)}])
+
+      await subscriber.initialize()
+
+      const message = { first: 'one'}
```
Comment (ddadonJoin) [inline packages/pubsub/src/__tests__/Subscriber.test.ts:456]: Same here on the message

**Review by ddadonJoin — COMMENTED** [review]
🔥💪 Fantastic work

File: packages/pubsub/src/__tests__/Subscriber.test.ts (hunk 1 above)
Comment (eugeneswave) [reply packages/pubsub/src/__tests__/Subscriber.test.ts]: fixed

File: packages/pubsub/src/AvroParser.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    return arraysPath
+  }
+
+  private findOptionalArrayPaths(field: Field, path: string[], arraysPath: string[]) {
+    let optional = false
+    let type = field.type
+    if (type instanceof types.UnwrappedUnionType) {
+      const typeAfterNull = type.types[1]
+      if (!typeAfterNull) {
+        throw Error('Second type should be always a real type, as union used only for optional values')
```
[the remark this answers is not in the document]
Comment (eugeneswave) [reply packages/pubsub/src/AvroParser.ts]: fixed

**Pull request merged**